src/main.rs

```diff
@@ -13,18 +17,64 @@ fn main() {
     }
 }
 
-fn run(mut args: impl Iterator<Item = String>) -> Result<()> {
-    while let Some(arg) = args.next() {
-        if arg == "--" {
-            break;
-        }
+
+async fn relay_signals(child: i32) -> Result<()> {
+    // All termination signals that can be caught
+    let mut signals : Signals = Signals::new(&[
+        Signal::Term,
+        Signal::Int,
+        Signal::Quit,
+        Signal::Hup,
+    ])?;
+
+    while let Some(signal) = signals.next().await {
+      let kill = Command::new("kill")
+          .args(["-s", &(signal.unwrap() as i32).to_string(), &child.to_string()])
+          .stderr(Stdio::null())
+          .stdout(Stdio::null())
+          .spawn()?.wait().await;
+      match kill {
+          Ok(_) => { continue; }
+          Err(_) => { break; }
+      }
+
     }
-    let command = args.next().context("comment: no command given")?;
-    let command_args: Vec<String> = args.collect();
-    Command::new(command).args(command_args).spawn()?.wait()?;
     Ok(())
 }
 
+fn run(mut args: impl Iterator<Item = String>) -> Result<()> {
+    tokio::runtime::Builder::new_multi_thread()
```

Comment:
If we do decide to keep tokio we should probably just make `run` be async and tag main with `#[tokio::main]` which is equivalent to this.